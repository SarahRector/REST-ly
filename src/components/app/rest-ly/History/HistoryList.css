.History {
  width: 20px;
}