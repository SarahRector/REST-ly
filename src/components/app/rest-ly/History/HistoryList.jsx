/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';
//import styles from '../History/HistoryList';

const HistoryList = ({ histories }) => {
  const eachHistory = histories.map(history => (
    <li key={`${history.method}+${history.url}`}>
      <HistoryItem method={history.method} url={history.url} body={history.body} />
    </li>
  ));

  return (
    <ul>
      {eachHistory}
    </ul>
  );
};

HistoryList.propTypes = {
  histories: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    body: PropTypes.string,
    method: PropTypes.string.isRequired
  })).isRequired
};

export default HistoryList;

