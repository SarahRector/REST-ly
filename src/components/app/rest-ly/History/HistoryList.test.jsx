/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryList from './HistoryList';

describe('HistoryList component', () => {
  afterEach(() => cleanup());

  const testArray = [
    { url: 'https://hey-arnold-api.herokuapp.com/api/v1/characters/random?count=2', method: 'GET' },
    { url: 'https://hey-arnold-api.herokuapp.com/api/v1/characters/random?count=2', method: 'POST' },
  ];

  it('renders HistoryList', () => {
    const { asFragment } = render(<HistoryList 
      histories={testArray}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
