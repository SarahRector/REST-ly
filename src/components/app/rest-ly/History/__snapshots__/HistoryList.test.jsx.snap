// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component renders HistoryList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <article>
        <p>
          GET
        </p>
        <span>
          https://hey-arnold-api.herokuapp.com/api/v1/characters/random?count=2
        </span>
      </article>
    </li>
    <li>
      <article>
        <p>
          POST
        </p>
        <span>
          https://hey-arnold-api.herokuapp.com/api/v1/characters/random?count=2
        </span>
      </article>
    </li>
  </ul>
</DocumentFragment>
`;
